kimchi/src/circuits/polynomials/foreign_field_add/gadget.rs

```diff
@@ -27,31 +30,40 @@ impl<F: PrimeField + SquareRootField> CircuitGate<F> {
     ///      [n+1]         -> 1 Zero row for bound result
     /// ]
     ///
-    pub fn create(
+    /// INTEGRATION:
+    /// - Wire the range check for result bound manually
+    /// - Connect to public input containing the 1 value for the overflow in the final bound check
+    pub fn create_chain_ffadd(
         start_row: usize,
-        num: usize,
+        opcodes: &[FFOps],
         foreign_field_modulus: &BigUint,
     ) -> (usize, Vec<Self>) {
         let next_row = start_row;
         let foreign_field_modulus = foreign_field_modulus.to_field_limbs::<F>();
         let mut circuit_gates = vec![];
-
+        let num = opcodes.len();
+        // ---------------------------
         // Foreign field addition gates
         // ---------------------------
         // First the single-addition gates
         for i in 0..num {
+            assert_ne!(opcodes[i], FFOps::Mul);
+            let mut coeffs = foreign_field_modulus.to_vec();
+            coeffs.push(opcodes[i].sign::<F>());
             circuit_gates.append(&mut vec![CircuitGate {
                 typ: GateType::ForeignFieldAdd,
                 wires: Wire::for_row(next_row + i),
-                coeffs: foreign_field_modulus.to_vec(),
+                coeffs,
             }]);
         }
+        let mut final_coeffs = foreign_field_modulus.to_vec();
+        final_coeffs.push(FFOps::Add.sign::<F>());
         // Then the final bound gate and the zero gate
         circuit_gates.append(&mut vec![
             CircuitGate {
                 typ: GateType::ForeignFieldAdd,
                 wires: Wire::for_row(next_row + num),
-                coeffs: foreign_field_modulus.to_vec(),
+                coeffs: final_coeffs,
```

Comment:
I would just put `foreign_field_modulus.to_vec()` here and avoid temporary, single purpose variable.

Reply:
but I don't just need foreign_field_modulus there, but that plus the sign coefficient